### ISSUE TITLE
Bug: [no-unused-vars] Type predicate parameter not reported as unused when only used in type position

### ISSUE THREAD

**Issue by camc314** [typescript-eslint/typescript-eslint#12003]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.9.2&fileType=.tsx&code=FAUwHgDg9gTgLgAgMZQHYGdEFsCeAVHCEAcQFcBDGAEwQF4EAKK8ucgLgVNQGtUoB3VAEoOzVggCW6BJhgTUAczoA%2BBAG9gCBDBBxSMVAjgxSIYAF8A3MFCRYiFBmz5CJCtQBMdRmPIeOXLwCwirqmtq6%2BobGphaWQA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1tidmUQAmtAG4BDaKgwBtSImjQO0SABpMUcQHNJ6%2BPEhgAvgF1w2AyANA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
// This should report `data` as unused (only used in type position) but doesn't                                  
export const myTypeGuard = (data: unknown): data is string => {                                                  
  return true                                                                                                    
};
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "@typescript-eslint/no-unused-vars": ["error"],
  },
};
```

### tsconfig

```jsonc

```

### Expected Result

I would expect `data` to be unused`

### Actual Result

no error is reported by the rule

### Additional Info

Also relevant:
```ts
// This should report `data` as unused (only used in type position) but doesn't                                  
export const myTypeGuard = (data: unknown): data is string => {                                                  
  return true                                                                                                    
};                                                                                                               
```
```ts
// This correctly reports `data2` as unused with message about type position                                     
export const myTypeGuard2 = (data2: unknown): typeof data2 => {                                                  
  return true                                                                                                    
};  
```

Discord discussion: https://discord.com/channels/1026804805894672454/1451903355440992307
